peachjam/views/documents.py

```diff
@@ -9,6 +11,55 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {"code": "aa", "domain": "agp.africanlii.org"},
+        "ghalii": {"code": "gh", "domain": "new.ghalii.org"},
+        "lawlibrary": {"code": "za", "domain": "lawlibrary.org.za"},
+        "lesotholii": {"code": "ls", "domain": "new.lesotholii.org"},
+        "malawilii": {"code": "mw", "domain": "malawilii.org"},
+        "namiblii": {"code": "na", "domain": "namiblii.org"},
+        "senlii": {"code": "sn", "domain": "senlii.org"},
+        "seylii": {"code": "sc", "domain": "new.seylii.org"},
+        "sierralii": {"code": "sl", "domain": "new.sierralii.org"},
+        "tanzlii": {"code": "tz", "domain": "new.tanzlii.org"},
+        "tcilii": {"code": "tc", "domain": "new.tcilii.org"},
+        "ulii": {"code": "ug", "domain": "new.ulii.org"},
+        "zambialii": {"code": "zm", "domain": "new.zambialii.org"},
+        "zanzibarlii": {"code": "tz-znz", "domain": "new.zanzibarlii.org"},
+        "zimlii": {"code": "zw", "domain": "zimlii.org"},
+    }
+
+    def __init__(self):
+        self.current_authority = self.RESOLVER_MAPPINGS[
+            settings.PEACHJAM["APP_NAME"].lower()
+        ]
+
+    def get_url_for_frbr_uri(self, frbr_uri):
+        try:
+            parsed_uri = FrbrUri.parse(frbr_uri)
+            country_code = parsed_uri.country
+            locality = parsed_uri.locality
+
+            # special case for tanzlii and zanzibarlii. All others are authoritative for their countries
+            if (
+                country_code == "tz"
+                and locality == "znz"
+                and self.current_authority["domain"]
+                != self.RESOLVER_MAPPINGS["zanzibarlii"]["domain"]
+            ):
+                return f"http://{self.RESOLVER_MAPPINGS['zanzibarlii']['domain']}{frbr_uri}"
+
+            if country_code != self.current_authority["code"]:
+                for k, v in self.RESOLVER_MAPPINGS.items():
+                    if country_code == v["code"]:
+                        return f"http://{v['domain']}{frbr_uri}"
+            return None
+        except ValueError:
```

Comment:
It's better to fail loudly than silently. I think this exception block is too broad -- if the code assumes something is in the mappings settings, it's better to fail when it's not than to silently do nothing.